src/System.Reflection/tests/MethodInfo/MethodInfo_CreateDelegateTests.cs

```diff
@@ -82,48 +77,40 @@ public void TestCreateDelegate7()
         {
             ClassA TestClass = (ClassA)Activator.CreateInstance(typeof(ClassA));
             MethodInfo miPublicInstanceMethod = GetMethod(typeof(ClassA), "PublicInstanceMethod");
-            ClassA classAObj = new ClassA();
             Assert.Throws<ArgumentNullException>(() =>
             {
                 miPublicInstanceMethod.CreateDelegate(null, TestClass);
             });
         }
 
-        //closed instance delegate with incorrect object type
-        [Fact]
-        public void TestCreateDelegate8()
+        public static IEnumerable<object[]> TestCreateDelegateData()
         {
-            ClassA TestClass = (ClassA)Activator.CreateInstance(typeof(ClassA));
-            MethodInfo miPublicInstanceMethod = GetMethod(typeof(ClassA), "PublicInstanceMethod");
-            ClassA classAObj = new ClassA();
-            Assert.Throws<ArgumentException>(() =>
-            {
-                miPublicInstanceMethod.CreateDelegate(typeof(Delegate_Void_Int), new DummyClass());
-            });
+            //closed instance delegate with incorrect object type
+            yield return new object[] { typeof(Delegate_Void_Int) };
+            //create closed static method with incorrect argument
+            yield return new object[] { typeof(Delegate_Void_Str) };
         }
 
-        //create closed static method with incorrect argument
-        [Fact]
-        public void TestCreateDelegate9()
+        [Theory]
+        [MemberData(nameof(TestCreateDelegateData))]
+        public void TestCreateDelegate(Type testData)
         {
-            ClassA TestClass = (ClassA)Activator.CreateInstance(typeof(ClassA));
             MethodInfo miPublicInstanceMethod = GetMethod(typeof(ClassA), "PublicInstanceMethod");
-            ClassA classAObj = new ClassA();
             Assert.Throws<ArgumentException>(() =>
             {
-                miPublicInstanceMethod.CreateDelegate(typeof(Delegate_Void_Str), new DummyClass());
+                miPublicInstanceMethod.CreateDelegate(testData, new DummyClass());
```

Comment:
Nit: it's ok as-is, but if you wanted, you could consolidate this to a single line, e.g.
```C#
Assert.Throws<ArgumentException>(() => miPublicInstanceMethod.CreateDelegate(testData, new DummyClass());
```
Also, see my comment about argument exceptions and Assert.Throws here: https://github.com/dotnet/corefx/pull/8609#discussion_r63582916